californium-core/src/main/java/org/eclipse/californium/core/network/Matcher.java

```diff
@@ -174,6 +174,10 @@ public void sendRequest(Exchange exchange,final  Request request) {
 		if (request.getToken() == null) {
 			idByToken = createUnusedToken();
 			request.setToken(idByToken.token);
+			// if original request has no token set it too.
+			// this is the case where request and currentRequest is not the same
+			if (exchange.getRequest() != null && exchange.getRequest().getToken() == null)
```

Comment:
`request` parameter is the current request, while `exchange.getRequest` is the original request.
Since I add this [modification](https://github.com/eclipse/californium/pull/16/files#diff-20c9d7fed566d6ad9272e84175d7b9c2R459), we could encounter some NPE because `exchange.getRequest.getToken` is `null`.

This is mainly the case where you send a request with blockwise transfer (block1).
In this case a new request is created in the `blockwiselayer` and so when we are in the `matcher` and create a new token for this request we need to modify the original one too.